src/utils/EntryModel.hxx

```diff
@@ -41,389 +41,465 @@
-    unsigned digits_     : 5; /**< number of significant digits */
-    unsigned index_      : 5; /**< present write index */
-    unsigned hasInitial_ : 1; /**< has an initial value */
-    unsigned transform_  : 1; /**< force characters to be upper case */
-    unsigned reserved_   : 20; /**< reserved bit space */
+    T value_; ///< present value held
+    T valueMin_; ///< minimum value representable by maxSize_
+    T valueMax_; ///< maximum value representable by maxSize_ 
 
-    int base_; /**< radix base */
-    char data_[N + 1]; /**< data string */
+    unsigned numLeadingZeros_  : 5; ///< number of leading zeros
+    unsigned maxSize_          : 5; ///< maximum number of digits
+    unsigned size_             : 5; ///< actual number of digits
+    unsigned isAtInitialValue_ : 1; ///< true if still has the initial value
+    unsigned empty_            : 1; ///< true if the value_ is "empty"
+    unsigned upper_            : 1; ///< force characters to be upper case
+    unsigned reserved_         : 15; ///< reserved bit space
+
+    int base_; ///< radix base
 
     DISALLOW_COPY_AND_ASSIGN(EntryModel);
 };
 
-/** Specialization of EntryModel with upper and lower bounds
- * @tparam T the data type up to 64-bits in size
- * @tparam N the size of the entry in max number of visible digits.
- */
-template <class T, size_t N> class EntryModelBounded : public EntryModel<T, N>
+/// Specialization of EntryModel with upper and lower bounds
+/// @tparam T the data type up to 64-bits in size
+/// @tparam N the size of the entry in max number of visible digits.
+template <class T> class EntryModelBounded : public EntryModel<T>
 {
 public:
-    /** Constructor.
-     * @param transform force characters to be upper case
-     */
-    EntryModelBounded(bool transform = false)
-        : EntryModel<T, N>(transform,
-                           std::bind(&EntryModelBounded::clamp, this))
+    /// Constructor.
+    /// @param upper force characters to be upper case
+    EntryModelBounded(bool upper = false)
+        : EntryModel<T>(upper)
     {
     }
 
-    /** Initialize with a value.
-     * @param digits max number of significant digits in the base type
-     * @param base base type, 10 or 16
-     * @param value unsigned value to initialize with
-     * @param min minumum value
-     * @param max maximum value
-     * @param default_val default value
-     */
-    void init(unsigned digits, int base, T value, T min, T max, T default_val)
-    {
-        min_ = min;
-        max_ = max;
-        default_ = default_val;
-        EntryModel<T, N>::init(digits, base, value);
-    }
-
-    /// Set the value to the minimum.
-    void set_min()
-    {
-        EntryModel<T, N>::set_value(min_);
-    }
-
-    /// Set the value to the maximum.
-    void set_max()
+    /// Initialize with a value.
+    /// @param max_size max number of digits in the base type
+    /// @param base base type, 10 or 16
+    /// @param value unsigned value to initialize with
```

Comment:
i think this does not need to be unsigned, it could be signed.

Reply:
Fixed.